recipes/default.rb

```diff
@@ -15,14 +15,14 @@
   node['rackspace_users']['data_bag'],
   node['rackspace_users']['data_bag_item']).to_hash.select { |user, user_data| user != 'id' }
 
-node_tags = node['rackspace_users']['node_tags']
+node_groups = node['rackspace_users']['node_groups']
 
 groups = {}
 
 users.each do |username, user_data|
-  ## The action (if defined in the data bag) should be overwritten to 'remove' if the user's node tags and node's tags don't intersect.
-  ## If the user doesn't declare any tags then it is assumed that is present on all nodes.
-  user_data['action'] = 'remove' if user_data['node_tags'] && (user_data['node_tags'] & node_tags).empty?
+  ## The action (if defined in the data bag) should be overwritten to 'remove' if not all the user's node_groups are in node's node_groups.
+  ## If the user doesn't declare node_groups then it is assumed that is present on all nodes.
+  user_data['action'] = 'remove' if user_data['node_groups'] && !(user_data['node_groups'] - node_groups).empty?
```

Comment:
To be honest, this feels like it's starting to get into way too specific an implementation. Our goal is to satisfy the 80%, and let the others adjust the template. As we're discovering, it's a bit more tricky to do this if we encapsulate all the logic into a cookbook (we learned this lesson with stacks, as well) vs. exposing the choices directly to the caller. 

I think we should go ahead and incorporate this code directly into a template, so it can be adjusted by the end user, -or- (my preferred solution would be) a library, which would really excel at this task. You could just have the user supply a block if they don't like the matching, e.g.:

Users must have at least one match against node groups and user have groups defined:
```
rackspace_users 'default' do
  access_check do |node_groups, user_groups|
    user_groups &&
      !(node_groups & user_groups).empty? # sets must have at least one intersecting member
  end
end
```

Users must match every node group and user have groups defined:
```
rackspace_users 'default' do
  access_check do |node_groups, user_groups|
    user_groups && 
      !node_groups.empty? && # sets must not be empty
      !users_groups.empty? && # sets must not be empty
      (node_groups - user_groups).empty? # sets must have all the same members
  end
end
```

Put all users everywhere or could be default:
```
rackspace_users 'default' do
  access_check { |_a, _b| true } # don't care
end
```